### DISCUSSION TITLE
Reproducible memory leak measurement

### DISCUSSION THREAD

**Discussion by jamesdbrock** [bloomberg/memray#570] [Q&A]
I have a program that runs for some time then stops. I want to measure the total memory leakage in that program in a reproducible way, so that I know if changes to the program are improvements.

I want to script the production of a number named **total_memory_leak_size** with these properties:
1. If **total_memory_leak_size** goes down, then I know my program is leaking less memory.
2. If **total_memory_leak_size** goes to zero, then I know my program is leaking zero memory.

What is the best way to produce **total_memory_leak_size** with `memray`? (Is there a command that's like `memray flamegraph --leaks` but produces **total_memory_leak_size** instead of an HTML page?)

**Comment by pablogsal** [comment]
Although for now it's not a public, stable interface you can do:

```
from memray import FileReader

reader = FileReader("myfile.bin")
leaked = sum(record.size for record in reader.get_leaked_allocation_records())
print(leaked)
```
